build/transpile.js

```diff
@@ -665,7 +665,7 @@ class Transpiler {
             [ /\}\s+catch \(([\S]+)\) {/g, '} catch (Exception $$$1) {' ],
             [ /for\s+\(([a-zA-Z0-9_]+)\s*=\s*([^\;\s]+\s*)\;[^\<\>\=]+(\<=|\>=|<|>)\s*(.*)\.length\s*\;([^\)]+)\)\s*{/g, 'for ($1 = $2; $1 $3 count($4);$5) {' ],
             [ /for\s+\(([a-zA-Z0-9_]+)\s*=\s*([^\;\s]+\s*)\;[^\<\>\=]+(\<=|\>=|<|>)\s*(.*)\s*\;([^\)]+)\)\s*{/g, 'for ($1 = $2; $1 $3 $4;$5) {' ],
-            [ /([^\s]+)\.length\;/g, 'count($1);' ],
+            [ /([^\s]+)\.length\;/g, 'is_array($1) ? count($1) : strlen($1);' ],
```

Comment:
I would remove this edit, we have a heuristic to prevent these issues and adding this conditional might have unexpected side-effects

Reply:
Okay but if .length is used on a string without a logical operator it doesn't compile to php properly because it will think it's an array and use `count` instead of `strlen` wouldn't it be good to amend that? Or are you saying that is usally prevented somehow? These changes that I made are similar to what was here before this edit https://github.com/ccxt/ccxt/pull/14580/files but it uses strlen instead of 0